x/eibc/types/lp.go

```diff
@@ -0,0 +1,69 @@
+package types
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dymensionxyz/gerr-cosmos/gerrc"
+)
+
+func (d OnDemandLP) Addr() (sdk.AccAddress, error) {
+	return sdk.AccAddressFromBech32(d.FundsAddr)
+}
+
+func (d OnDemandLP) MustAddr() sdk.AccAddress {
+	a, err := d.Addr()
+	if err != nil {
+		panic(err)
+	}
+	return a
+}
+
+func (d OnDemandLP) Validate() error {
+	if _, err := d.Addr(); err != nil {
+		return errorsmod.Wrap(err, "addr")
+	}
+	if err := validateRollappID(d.Rollapp); err != nil {
+		return errorsmod.Wrap(err, "rollapp id")
+	}
+	if sdk.ValidateDenom(d.Denom) != nil {
+		return errorsmod.Wrap(gerrc.ErrInvalidArgument, "denom")
+	}
+	if d.MaxPrice.IsNil() || !d.MaxPrice.IsPositive() {
```

Comment:
why is maxPrice mandatory?

Reply:
I don't see how INF or whatever is simpler than just not passing anything. In `FulfillOrderAuthorized` MaxPrice is optional, so it would be nice to have some consistency here

Reply:
Because if you make zero a special value you add branching to the code
More branching more problems, more testing required, no benefit

Reply:
Just do it on the frontend what's the big deal